pkg/datastore/deploymentchainstore.go

```diff
@@ -51,6 +53,31 @@ var (
 			return nil
 		}
 	}
+
+	DeploymentChainNodeDeploymentStatusUpdater = func(blockIndex uint32, deploymentID string, status model.DeploymentStatus, reason string) func(*model.DeploymentChain) error {
+		return func(dc *model.DeploymentChain) error {
+			if blockIndex >= uint32(len(dc.Blocks)) {
+				return fmt.Errorf("invalid block index provided")
+			}
+			block := dc.Blocks[blockIndex]
+			var updated bool
+			for _, node := range block.Nodes {
+				if node.DeploymentRef.DeploymentId != deploymentID {
+					continue
+				}
+				if node.DeploymentRef == nil {
+					continue
+				}
```

Comment:
These checks should be swapped up and down to prevent it from panic.

Reply:
Nice catch 👍